src/coord/src/coord.rs

```diff
@@ -207,29 +203,36 @@ where
                         .map(|src| (src.variant.clone(), src.index_id))
                         .collect(),
                 }),
-            );
+            )
+            .await;
         }
 
-        let (persister, persistence_tx, persistence_path) = if let Some(persistence_config) =
-            config.persistence
-        {
-            let (persistence_tx, persistence_rx) = config.switchboard.mpsc();
-            broadcast(
-                &mut broadcast_tx,
-                SequencedCommand::EnablePersistence(persistence_tx.clone()),
-            );
-            let path = persistence_config.path.clone();
-            (
-                Some(Persister::new(persistence_rx, persistence_config)),
-                Some(block_on(persistence_tx.connect()).expect("failed to connect persistence tx")),
-                Some(path),
-            )
-        } else {
-            (None, None, None)
-        };
+        let (persister, persistence_tx, persistence_path) =
+            if let Some(persistence_config) = config.persistence {
+                let (persistence_tx, persistence_rx) = config.switchboard.mpsc();
+                broadcast(
+                    &mut broadcast_tx,
+                    SequencedCommand::EnablePersistence(persistence_tx.clone()),
+                )
+                .await;
+                let path = persistence_config.path.clone();
+                (
+                    Some(Persister::new(persistence_rx, persistence_config)),
+                    Some(
+                        persistence_tx
+                            .connect()
+                            .await
+                            .expect("failed to connect persistence tx"),
+                    ),
+                    Some(path),
+                )
+            } else {
+                (None, None, None)
+            };
 
         let mut coord = Self {
             switchboard: config.switchboard,
+            cmd_rx: Some(config.cmd_rx),
```

Comment:
nit / out of scope for this change: theres a number of things in the Coordinator that are options only because we `take` them in `serve` and do stuff with them. maybe a better pattern is to have one option struct of stuff thats needed for serving that we take? this way is a bit confusing to read

Reply:
100% I hate this too. One step closer with not having `Coord` public though!